lib/Resolver.js

```diff
@@ -78,6 +78,19 @@ Resolver.prototype.resolve = function resolve(context, path, request, callback)
 	}, null));
 };
 
+Resolver.prototype.doResolveWithErrorLogging = function doResolveWithErrorLogging(type, request, message, callback, options) {
+	var optionsWithoutLog = {
+		stack: options.stack,
+		missing: options.missing
+	};
+
+	this.doResolve(type, request, message, createInnerCallback((err, result) => {
```

Comment:
basically, it returns the old function after 2 conditionals if everything is OK, so it's extremely low overhead. Avoiding the GC is the huge win you discovered. I think we can afford the 2 correctness checks for now, and if it still remains a perf issue, address in a follow up... thoughts?